src/creature.cpp

```diff
@@ -460,32 +461,32 @@ void Creature::onDeath()
 {
 	bool lastHitUnjustified = false;
 	bool mostDamageUnjustified = false;
-	Creature* lastHitCreature = g_game.getCreatureByID(lastHitCreatureId);
-	Creature* lastHitCreatureMaster;
+	std::shared_ptr<Creature> lastHitCreature = g_game.getCreatureByID(lastHitCreatureId);
```

Comment:
Could this be `const auto`
```suggestion
	const auto lastHitCreature = g_game.getCreatureByID(lastHitCreatureId);
```